### ISSUE TITLE
trident-operator fails to patch trident-csi ClusterRole in 20.10.0 upgrade

### ISSUE THREAD

**Issue by adrianchifor** [NetApp/trident#491]
**Describe the bug**

We recently upgraded from 20.07.1 to 20.10.0 and noticed all volume attachments stopped working. The operator failed to patch the trident-csi ClusterRole to include `volumeattachments/status` RBAC: https://github.com/NetApp/trident/commit/260c6ab457cca5149dd142f1d4d6ea23ed16e346

However the operator logs don't mention any errors:
```
time="2020-12-07T12:00:38Z" level=info msg="A Trident cluster role named 'trident-csi' was found by label."
time="2020-12-07T12:00:38Z" level=debug msg="Patching Trident Cluster role." clusterRole=trident-csi
```

I believe the current logic is flawed, as it doesn't check the patch status/error:
https://github.com/NetApp/trident/blob/master/operator/controllers/provisioner/installer/installer.go#L969

**Environment**

- Trident version: 20.10.0
- Trident installation flags used: --debug
- Container runtime: cri-o://1.19.0-24
- Kubernetes version: v1.19.0
- Kubernetes orchestrator: OpenShift v4.6
- OS: Red Hat Enterprise Linux CoreOS 46.82

**To Reproduce**
Install trident 20.07.1 and upgrade to 20.10.0. Check if trident-csi ClusterRole was patched with `volumeattachments/status` permission.

**Expected behavior**
Operator should patch the trident-csi ClusterRole, handle any errors and log accordingly to help the user debug and fix the issue.

**Comment by adrianchifor** [comment]
Opened https://github.com/NetApp/trident/pull/492 for better debugging

**Comment by ntap-arorar** [comment]
Hello @adrianchifor 

Thanks for bringing this to attention, certainly there is a need to add a log statement after the patch and return `err` instead of nil. Would appreciate if you share more details on how you ran into this issue.

1. Did you run into this issue during an upgrade or post-upgrade?
2. Do you have the error that was thrown during the patch operation?
3. Are you using `tridentImage` attribute in your CR to set a custom image?

The reason I am asking this is that, during upgrade from 20.07.* to 20.10.0, all the Trident objects (except for the CRDs) are deleted and re-created and not patched. However, if you have a `tridentImage` attribute in your CR the `imagePrechecks` run to verify that you have supplied a valid Trident image with the correct Trident version, and that is when cluster-role patch may have been called. `imagePrechecks` is a step just before the upgrade and only runs when you have supplied a custom image via `tridentImage`.

**Comment by adrianchifor** [comment]
1. post-upgrade
2. no error as it's not handled
3. custom image yes (our docker.io mirror)

**Comment by gnarl** [comment]
This is fixed with [commit a0712e](https://github.com/NetApp/trident/commit/a0712ef56008f55b2b6f3db06055548d634b93ed) and is included in the Trident v21.01 release.
